app/controllers/application_controller.rb

```diff
@@ -9,15 +9,22 @@ def authorize_request
     begin
       @decoded = AuthTokenService.decode(header)
       @current_user = User.find(@decoded['user_id'])
+      ownership(@current_user)
     rescue ActiveRecord::RecordNotFound => e
       render json: { errors: e.message }, status: :unauthorized
     rescue JWT::DecodeError => e
       render json: { errors: e.message }, status: :unauthorized
     end
   end
+
+  def ownership(current_user)
+    token_id = current_user.id
+    if token_id != params[:id].to_i && current_user.role.name != 'admin'
```

Comment:
Por qué la condición AND? No bastaría con que alguno de los dos no se cumpla para que ya no sea válido?

Reply:
Tendría que ser un OR entonces?